File: keeper/Models/Keep.cs
namespace keeper.Models
{
  public class Keep : IDbItem<int>, IHasCreator
  {
    public int Id { get; set; }
    public string CreatorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public int Views { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // NOTE Virtuals
    public Profile Creator { get; set; }
    public int kept { get; set; }
  }
}

File: keeper.client/src/models/Keep.js
import { Profile } from "./Profile.js";

export class Keep {
  constructor(data) {
    this.id = data.id;
    this.creatorId = data.creatorId;
    this.name = data.name;
    this.description = data.description;
    this.img = data.img;
    this.views = data.views;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
    this.creator = new Profile(data.creator);
    this.kept = data.kept;
  }
}
